### PR TITLE
added executor_threads documentation

### PR DESCRIPTION
**Pull request by 111andre111**
Thanks for contributing to Logstash! If you haven't already signed our CLA, here's a handy link: https://www.elastic.co/contributor-agreement/

### CODE REVIEW

**Comment by 111andre111** [comment]
Here was a parallel one:
https://github.com/logstash-plugins/logstash-input-beats/pull/409

File: docs/index.asciidoc [hunk 1]
```diff
@@ -170,6 +171,14 @@ The list of ciphers suite to use, listed by priorities.
 
 Close Idle clients after X seconds of inactivity.
 
+[id="plugins-{type}s-{plugin}-executor_threads"]
+===== `executor_threads`
+
+  * Value type is <<number,number>>
+  * Default value is 1 executor thread per CPU core
+
+The number of parallel executor threads.
```
Comment (andsel) [inline docs/index.asciidoc:180]:
```suggestion
The number of parallel executor threads to be used to process incoming beats requests
```

**Review by andsel — COMMENTED** [review]
Just left a minimal suggestion

**Comment by 111andre111** [comment]
@andsel 
What is the influcence of the draft PR https://github.com/logstash-plugins/logstash-input-beats/pull/410
mentioned in  https://github.com/logstash-plugins/logstash-input-beats/pull/409?

**Comment by andsel** [comment]
that description is more extended, we could use that, and close that draft PR with this one

**Comment by andsel** [comment]
the #410 is some more deep, where the #409 and this are only a face of the problem it's trying to fix

**Comment by andsel** [comment]
Hi @111andre111 I've rebased to `master` branch fixing conflicts and improved description.

**Comment by 111andre111** [comment]
So I assume we can merge this one and #409 as well if there weren't duplicates plus  I assume #410 still will need some investigation time, correct? @andsel

**Comment by andsel** [comment]
Yes correct @111andre111, but Changelog needs a fix

**Comment by 111andre111** [comment]
@andsel I have fixed now the Changelog

File: docs/index.asciidoc [hunk 2]
```diff
@@ -224,6 +225,24 @@ Close Idle clients after X seconds of inactivity.
...
+[id="plugins-{type}s-{plugin}-executor_threads"]
+===== `executor_threads`
+
+  * Value type is <<number,number>>
+  * Default value is 1 executor thread per CPU core
+
+The number of threads to be used to process incoming beats requests.
+By default Beats input will create a number of threads equals to 2*CPU cores to handle incoming connections,
+reading from the established sockets and execute most of the tasks related to network connection managements, 
+except the parsing of Lumberjack protocol that's offloaded to a dedicated thread pool.
```
Comment (karenzone) [inline docs/index.asciidoc:237]:
```suggestion
By default the Beats input creates a number of threads equal to 2*CPU cores.
These threads handle incoming connections, reading from established sockets, and executing most of the tasks related to network connection management. 
Parsing the Lumberjack protocol is offloaded to a dedicated thread pool.
```

**Review by karenzone — COMMENTED** [review]
Thank you for adding this important documentation. Nice job explaining it. I left some minor suggestions inline for your consideration.

File: docs/index.asciidoc (hunk 2 above)
Comment (111andre111) [reply docs/index.asciidoc:237]: Nice, thank you Karen for the additional suggestions.

**Review by andsel — APPROVED** [review]
LGTM, merge squashing the commits please, or ping me if you need I do it

**Comment by 111andre111** [comment]
@andsel  Sorry if I didn't manage the squash. I always got an error message.

**Comment by andsel** [comment]
@111andre111 I've rebased to `master` there were some conflicts on `CHANGELOG.md` ao probably that prohibited the merge.
I don't publish yet this, I left the version `x.y.z` so that a next bugfix or feature also publish this.

**Comment by 111andre111** [comment]
Ah I see. I didn't see that. Thank you @andsel

**Pull request merged**